settings.gradle.kts

```diff
@@ -6,3 +6,10 @@ include(":gr8-plugin")
 include(":gr8-plugin-external")
 include(":gr8-plugin-common")
 
+/**
+ * We need Java <= 17 until we update the embedded gr8
+ * See https://issuetracker.google.com/u/2/issues/365578411
+ */
+check(JavaVersion.current() <= JavaVersion.VERSION_17) {
```

Comment:
Should this use the Gradle toolchain mechanism instead?

Reply:
[Not the biggest toolchain fan](https://jakewharton.com/gradle-toolchains-are-rarely-a-good-idea/). What I really want is to use Java 23 but this requires https://issuetracker.google.com/u/2/issues/365578411

Reply:
> this requires https://issuetracker.google.com/u/2/issues/365578411

That's marked as fixed. But I guess there's no release with it yet?

Reply:
It should be released by now. What we need to do to update R8 in here and then make a new release of Gr8 and then use that new Gr8 release in `main` and then we can remove the limitation. Bootstrap all the way down 😅 

Note that in this specific case, toolchains wouldn't help anyways because R8 takes the runtime libs from the current JRE (unless we used a separate JavaExec task to run R8 itself but there are so many hours in a day sadly).

Reply:
I've created https://github.com/GradleUp/gr8/issues/23 to track this.